iterator/src/test/java/com/iluwatar/iterator/list/TreasureChestTest.java

```diff
@@ -60,17 +56,18 @@
   }
 
   /**
-   * Test if the expected item can be retrieved from the chest using the {@link ItemIterator}
+   * Test if the expected item can be retrieved from the chest using the {@link
+   * TreasureChestItemIterator}
    */
   @ParameterizedTest
   @MethodSource("dataProvider")
   public void testIterator(Item expectedItem) {
     final TreasureChest chest = new TreasureChest();
-    final ItemIterator iterator = chest.iterator(expectedItem.getType());
+    final Iterator iterator = chest.iterator(expectedItem.getType());
     assertNotNull(iterator);
 
     while (iterator.hasNext()) {
-      final Item item = iterator.next();
+      final Item item = (Item) iterator.next();
```

Comment:
No need for cast now, it was needed because of raw types. It will be removed now.